includes/class-sensei-course.php

```diff
@@ -3728,7 +3742,10 @@ public static function can_current_user_manually_enrol( $course_id ) {
 		// Check if the user is already enrolled through any provider.
 		$is_user_enrolled = self::is_user_enrolled( $course_id, get_current_user_id() );
 
-		$default_can_user_manually_enrol = is_user_logged_in() && ! $is_user_enrolled;
+		// Check if self enrollment is not allowed for this course.
+		$is_self_enrollment_not_allowed = self::is_self_enrollment_not_allowed( $course_id );
```

Comment:
Extra check for this "capability" method, which already solves many cases, like hiding buttons and preventing enrollment.